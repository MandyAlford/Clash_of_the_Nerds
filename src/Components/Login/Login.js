import React, { Component } from 'react';
import './Login.scss';

class Login extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      team: 'mythology'
    }
  }
  handleChange = (event) => {
    this.setState({ name: event.target.value })
  }
  submitInfo = (props) => {
    this.props.addUser({ name: this.state.name, team: this.state.team })
    this.setState({ name: '' })
  }
  render() {
    return(
      <form className='login-form'>
        <h1 className='title-logo'>Clash of the Nerds</h1>
        <input className='user-name' placeholder='name' type='text' value={this.state.name} onChange={this.handleChange} maxlength='25'/>
        <h3>Choose Your Team</h3>
        <select className='form-choice' onChange={(event) => this.setState({team: event.target.value})}>
          <option value='mythology'>Mythology</option>
          <option value='science'>Science & Nature</option>
          <option value='art'>Art</option>
        </select>
        <button type='button' className='submit-btn' onClick={this.submitInfo}>Ready to Rumble</button>
      </form>
    )
  }
}

export default Login
