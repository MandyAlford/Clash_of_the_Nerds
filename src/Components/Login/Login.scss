.login-form {
  background-color: rgba(255, 255, 255, 0.9);
  height: 350px;
  width: 550px;
  border-radius: 15px;
  z-index: 25;
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 20%;
  left: 30%;
  padding: 30px;
  border: 5px solid #662324;
  align-items: center;
}

.user-name {
  height: 40px;
  font-size: 25px;
  padding: 5px;
  margin-bottom: 20px;
  width: 400px;
}

.title-logo {
  color: #662324;
  margin-bottom: 20px;
}

.active {
  background-color: #662324;
  color: #FFF;
}

.form-choice {
  margin: 10px;
  font-size: 20px;
  height: 35px;
  border-radius: 5px;
  width: 400px;
  color: #662324;
}

.submit-btn {
  height: 40px;
  width: 200px;
  background-color: #662324;
  color: #FFF;
  font-size: 20px;
  margin-top: 30px;
}
