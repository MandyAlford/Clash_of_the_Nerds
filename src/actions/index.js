export const addUser = (user) => ({
    type: 'ADD_USER',
    user
})

export const addTrivia = (trivia) => ({
    type: 'ADD_TRIVIA',
    trivia
})

export const addCorrectGuess = (guess) => ({
  type: 'ADD_CORRECT_GUESS',
  guess
})

export const addIncorrectGuess = (guess) => ({
  type: 'ADD_INCORRECT_GUESS',
  guess
})

export const resetAnswers = () => ({
  type: 'RESET_STORE_ANSWERS'
})

export const saveUsersTrivia = (trivia) => ({
  type: 'SAVE_TRIVIA',
  trivia
})
