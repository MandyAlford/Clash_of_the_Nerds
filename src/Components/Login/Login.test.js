import React from 'react';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';
import { shallow } from 'enzyme';
import { addUser } from '../../actions';



describe('Login', () => {
  let wrapper, mockUser, mockFetch;

  beforeEach(() => {
    mockUser = {
      name: 'Jessica Rabbit',
      team: '19',
      level: 'hard'
    }
    mockFetch = jest.fn()
    wrapper = shallow(<Login
      fetchData={mockFetch}
      addUser={jest.fn()}
    />)
  })

  it('should match snapshot', () => {
    // const mockDispatch = jest.fn()
    // const actionToDispatch = addUser(mockUser)
    // const mappedProps = mapDispatchToProps(mockDispatch)
    // mappedProps.loadUserToStore(mockUser)
    // expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    expect(wrapper).toMatchSnapshot()
  })

})
