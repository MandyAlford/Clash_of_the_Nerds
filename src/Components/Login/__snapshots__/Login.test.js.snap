// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<form
  className="login-form"
>
  <img
    alt="clash of the nerds logo"
    className="title-logo"
    src="logo.png"
  />
  <input
    className="user-name"
    maxLength="25"
    onChange={[Function]}
    placeholder="name"
    type="text"
    value=""
  />
  <h3>
    Choose Your Battle
  </h3>
  <select
    className="form-choice"
    onChange={[Function]}
  >
    <option
      value="20"
    >
      Mythology
    </option>
    <option
      value="17"
    >
      Science & Nature
    </option>
    <option
      value="9"
    >
      General Knowledge
    </option>
    <option
      value="27"
    >
      Animals
    </option>
    <option
      value="19"
    >
      Mathematics
    </option>
  </select>
  <select
    className="form-choice"
    onChange={[Function]}
  >
    <option
      value="easy"
    >
      Easy
    </option>
    <option
      value="medium"
    >
      Medium
    </option>
    <option
      value="hard"
    >
      Hard
    </option>
  </select>
  <Link
    to="/test"
  >
    <button
      className="submit-btn"
      onClick={[Function]}
      type="button"
    >
      Ready to Rumble
    </button>
  </Link>
</form>
`;
